File: java/ChalkReplacer.java
class ChalkReplacer {
    public int whoReplacesIt(int[] chalk, int k) {
        final int MAX_STUDENTS = chalk.length;
        int index = -1;
        while (k >= 0) {
            index++;
            if (index >= MAX_STUDENTS)
                index = 0;
            k -= chalk[index];
        }
        return index;
    }

    public static void main(String[] args) {
        long startTime = System.nanoTime();
        ChalkReplacer chalkReplacer = new ChalkReplacer();
        int[] chalk = { 3, 4, 1, 2 };
        System.out.println(chalkReplacer.whoReplacesIt(chalk, 25));
        long endTime = System.nanoTime();
        double totalTime = (double) (endTime - startTime) / 1000000;
        System.out.println(totalTime + " milliseconds");
    }
}

File: python/chalkReplacer.py
from datetime import datetime


def chalkReplacer(chalk, k):
    maxStudents = len(chalk) - 1
    index = -1
    while k >= 0:
        index += 1
        if index > maxStudents:
            index = 0
        k -= chalk[index]
    return index


start_time = datetime.now()
print(chalkReplacer([3, 4, 1, 2], 25))
end_time = datetime.now()
time_diff = (end_time - start_time)
execution_time = time_diff.total_seconds() * 1000
print(f"{execution_time} milliseconds")
